CairoMakie/src/overrides.jl

```diff
@@ -35,15 +35,24 @@ function draw_poly(scene::Scene, screen::CairoScreen, poly, points::Vector{<:Poi
     draw_poly(scene, screen, poly, points, poly.color[], poly.model[], poly.strokecolor[], poly.strokewidth[])
 end
 
-function draw_poly(scene::Scene, screen::CairoScreen, poly, points::Vector{<:Point2}, color, model, strokecolor, strokewidth)
+# when color is a Makie.AbstractPattern, we don't need to go to Mesh
+function draw_poly(scene::Scene, screen::CairoScreen, poly, points::Vector{<:Point2}, color::Union{Symbol, Colorant, Makie.AbstractPattern},
+        model, strokecolor, strokewidth)
     space = to_value(get(poly, :space, :data))
     points = project_position.(Ref(scene), space, points, Ref(model))
     Cairo.move_to(screen.context, points[1]...)
     for p in points[2:end]
         Cairo.line_to(screen.context, p...)
     end
     Cairo.close_path(screen.context)
-    Cairo.set_source_rgba(screen.context, rgbatuple(to_color(color))...)
+    if color isa Makie.AbstractPattern
+        cairopattern = Cairo.CairoPattern(color)
+        Cairo.pattern_set_extend(cairopattern, Cairo.EXTEND_REPEAT);
```

Comment:
You should also set the filter so that Cairo doesn't interpolate...
```suggestion
        Cairo.pattern_set_extend(cairopattern, Cairo.EXTEND_REPEAT);
        Cairo.pattern_set_filter(cairopattern, Cairo.FILTER_NEAREST); # or something like this anyway

Reply:
not sure, why no interpolation? should make diagonal line patterns look better no

Reply:
Makes sense - was looking at it from the perspective of ImagePattern, where you may not want the pattern to be interpolated.  Maybe we could make this user-configurable somehow, perhaps by checking the `interpolate` keyword...